fastlane_core/lib/fastlane_core/tag_version.rb

```diff
@@ -0,0 +1,24 @@
+require "rubygems"
+
+module FastlaneCore
+  # Utility class to construct a Gem::Version from a tag.
```

Comment:
I think this is a pretty reasonable way to encapsulate this tag stuff in a reusable way. It seems like this might not be the only time you'd want to parse tags as versions using RubyGems. I'm out of time for this until this evening. I can add some tests for this class later.